inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/oceanbase/OceanBaseSourceOperator.java

```diff
@@ -0,0 +1,124 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.service.source.oceanbase;
+
+import org.apache.inlong.manager.common.consts.DataNodeType;
+import org.apache.inlong.manager.common.consts.InlongConstants;
+import org.apache.inlong.manager.common.consts.SourceType;
+import org.apache.inlong.manager.common.enums.ErrorCodeEnum;
+import org.apache.inlong.manager.common.exceptions.BusinessException;
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.common.util.JsonUtils;
+import org.apache.inlong.manager.dao.entity.StreamSourceEntity;
+import org.apache.inlong.manager.pojo.node.oceanbase.OceanBaseDataNodeInfo;
+import org.apache.inlong.manager.pojo.source.SourceRequest;
+import org.apache.inlong.manager.pojo.source.StreamSource;
+import org.apache.inlong.manager.pojo.source.oceanbase.OceanBaseBinlogSource;
+import org.apache.inlong.manager.pojo.source.oceanbase.OceanBaseBinlogSourceDTO;
+import org.apache.inlong.manager.pojo.source.oceanbase.OceanBaseBinlogSourceRequest;
+import org.apache.inlong.manager.pojo.stream.StreamField;
+import org.apache.inlong.manager.service.source.AbstractSourceOperator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Binlog source operator
+ */
+@Service
+public class OceanBaseSourceOperator extends AbstractSourceOperator {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public Boolean accept(String sourceType) {
+        return SourceType.OCEANBASE.equals(sourceType);
+    }
+
+    @Override
+    protected String getSourceType() {
+        return SourceType.OCEANBASE;
+    }
+
+    @Override
+    public String getExtParams(StreamSourceEntity sourceEntity) {
+        OceanBaseBinlogSourceDTO OceanBaseBinlogSourceDTO = JsonUtils.parseObject(sourceEntity.getExtParams(),
+                OceanBaseBinlogSourceDTO.class);
+        if (Objects.nonNull(OceanBaseBinlogSourceDTO) && StringUtils.isBlank(OceanBaseBinlogSourceDTO.getHostname())) {
+            OceanBaseDataNodeInfo dataNodeInfo = (OceanBaseDataNodeInfo) dataNodeService.get(
+                    sourceEntity.getDataNodeName(), DataNodeType.OCEANBASE);
+            CommonBeanUtils.copyProperties(dataNodeInfo, OceanBaseBinlogSourceDTO, true);
+            OceanBaseBinlogSourceDTO.setUser(dataNodeInfo.getUsername());
```

Comment:
Why not name the `user` in `OceanBaseBinlogSourceDTO` as `username`? so that `CommonBeanUtils.compyProperties `can help assign values.

Reply:
Revised